### ISSUE TITLE
Default empty arguments

### ISSUE THREAD

**Issue by seriouslag** [7nohe/openapi-react-query-codegen#37]
It would be nice if the arguments for the generated hooks had a default value if all the properties of the argument can be undefined.

Example of generated hook as is:
```typescript
export const useCatsServiceGetCats = <
  TData = CatServiceGetCatsDefaultResponse,
  TError = unknown,
  TQueryKey extends Array<unknown> = unknown[]
>(
  {
    color,
    age,
  }: {
    color?: string;
    age?: number;
  },
  queryKey?: TQueryKey,
  options?: Omit<
    UseQueryOptions<TData, TError>,
    "queryKey" | "queryFn" | "initialData"
  >
) =>
  useQuery<TData, TError>({
    queryKey: [
      useCatsServiceGetCatssKey,
      ...(queryKey ?? [{ color, age }]),
    ],
    queryFn: () =>
      CatsService.getCats(color, age) as TData,
    ...options,
  });
  
 // usage - need to pass empty object at least
 const {data} = useCatsServiceGetCats({});
```

Example of proposed generated code:
```typescript
export const useCatsServiceGetCats = <
  TData = CatServiceGetCatsDefaultResponse,
  TError = unknown,
  TQueryKey extends Array<unknown> = unknown[]
>(
  {
    color,
    age,
  }: {
    color?: string;
    age?: number;
  } = {}, // this is defaulted since all properties can be undefined
  queryKey?: TQueryKey,
  options?: Omit<
    UseQueryOptions<TData, TError>,
    "queryKey" | "queryFn" | "initialData"
  >
) =>
  useQuery<TData, TError>({
    queryKey: [
      useCatsServiceGetCatssKey,
      ...(queryKey ?? [{ color, age }]),
    ],
    queryFn: () =>
      CatsService.getCats(color, age) as TData,
    ...options,
  });
 
 // usage - no need to pass empty object
const {data} = useCatsServiceGetCats();
```

**Comment by 7nohe** [comment]
I also prefer that one 👍🏼

**Comment by seriouslag** [comment]
Closed by #60
